### ISSUE TITLE
Lower bound peak inconsistency

### ISSUE THREAD

**Issue by SomeUserName1** [fooof-tools/fooof#244]
Hi,

the following warning is very confusing to me as it contains a logically false statement:
```
FOOOF WARNING: Skipping frequency == 0, as this causes a problem with fitting.

FOOOF WARNING: Lower-bound peak width limit is < or ~= the frequency resolution: 97.66 <= 0.50
	Lower bounds below frequency-resolution have no effect (effective lower bound is the frequency resolution).
	Too low a limit may lead to overfitting noise as small bandwidth peaks.
	We recommend a lower bound of approximately 2x the frequency resolution.
```
Obviously 97.66 > 0.50. Maybe the error message is wrong and should be 99.76 > 12 and about the upper bound.

Context: 
I'm analyzing LFPs recorded of a slice stimulated chemically with NMDA and recorded with a 252-electrode MEA with an electrode distance of 200 mu.

Code to exec FOOOF:
```
def show_periodic_aperiodic_decomp(data):
    fms = []
    for idx in range(data.data.shape[0]):
        fms.append(compute_periodic_aperiodic_decomp(data.data[idx], data.sampling_rate))

    np.array(fms)
    names = get_selected_electrode_names(data)

    proc = Process(target=plot_periodic_aperiodic_decomp, args=(fms, names, len(data.selected_rows)))
    proc.start()

def compute_periodic_aperiodic_decomp(data_row, fs):
    fm = FOOOF(aperiodic_mode='knee')
    psd = sg.welch(data_row, fs=fs)
    fm.add_data(psd[0], psd[1])
    fm.fit()

    return fm
```

plotting code (for completeness):
```
def plot_periodic_aperiodic_decomp(fms, names, n_electrodes):
    use('TkAgg')

    grid_sz = int(np.ceil(np.sqrt(n_electrodes)))
    fig, ax_array = plt.subplots(grid_sz, grid_sz, sharey=True, sharex=True)
    if grid_sz == 1:
        ax_array = np.atleast_2d(ax_array)

    for i in range(grid_sz):
        if i * grid_sz >= n_electrodes:
            break

        for j in range(grid_sz):
            if i * grid_sz + j >= n_electrodes:
                break

            fms[i * grid_sz + j].plot(plot_aperiodic=True, plot_peaks='shade', plt_log=False, ax=ax_array[i, j])
            ax_array[i, j].legend(loc='upper right')

    fig.tight_layout()
    plt.show()
    plt.close()
```

Attached is a 5s snipped of an LFP, the corresponding PSD and what FOOOF spits out.

![fooof](https://user-images.githubusercontent.com/20174918/200024407-0841348a-edf2-4b0a-af92-64ce30752232.png)
![psd](https://user-images.githubusercontent.com/20174918/200024411-941dd787-9051-4813-b3a5-ea055ee73bbc.png)
![raw](https://user-images.githubusercontent.com/20174918/200024415-542e8afd-32d4-48eb-8f66-103cf7d8a253.png)

Should I adjust the upper limit?

Am I doing sth. else wrong?

Thanks in advance & cheers,
Fabian

**Comment by TomDonoghue** [comment]
Huh - something has gone a bit weird here - you are correct that this warning should not be reporting an obviously wrong comparison like this. It should be specifically about the lower peak range, so the value on the left should be frequency resolution, that it thinks is lower than the 0.5 peak width here. 

There is some kind of issue with computing / reporting the frequency resolution and how this relates to the peak width. It seems this relates to a frequency range that is somewhat unexpected - is a frequency range of 0-12000 correct? I'm assuming that's not the correct range - it's way broader than anything I've tried, and this range must be creating an issue somewhere, though exactly why it leads to this issue is something I would have to look deeper into.

Could you confirm what the frequency range is supposed to be, and if looking into that doesn't address it, could you send me a single psd (frequency array & power values), and I'll look into it when I get a chance? I think you can upload here - if not, email is `tdonoghue.research@gmail.com`.

**Comment by SomeUserName1** [comment]
The high fs is due to recording with a MEA system. Tried to convince the data recording (life science/mol. biology) guys that sampling with 2k-5k would be more than enough to capture most sub-threshold effects in populations, but I'm just a master student, so people only sometimes listen...

The PSD is computed with fs = 25k, i.e. the range is [0, 12.5k].

Tried after downsampling to 1k, i.e. a range of [0, 500]. (different recording & channels than last time though) 

Output:
```
FOOOF WARNING: Skipping frequency == 0, as this causes a problem with fitting.

FOOOF WARNING: Lower-bound peak width limit is < or ~= the frequency resolution: 3.91 <= 0.50
	Lower bounds below frequency-resolution have no effect (effective lower bound is the frequency resolution).
	Too low a limit may lead to overfitting noise as small bandwidth peaks.
	We recommend a lower bound of approximately 2x the frequency resolution.

Model fitting was unsuccessful.
```
![fooof](https://user-images.githubusercontent.com/20174918/200043727-6e2c8526-7180-4876-b262-b195c8ec7346.png)
![psd](https://user-images.githubusercontent.com/20174918/200043731-018f3e89-7797-433c-9b2a-dc9a7479e9a5.png)

> could you send me a single psd (frequency array & power values)

Will do on Monday. 

Thanks for your help so far & the fast reply, Dr. Donoghue!

**Comment by TomDonoghue** [comment]
Hey @SomeUserName1 - sorry for a long delay on getting back to you on this!

I checked through the data you sent, and figured out that the warning you noticed is wrong - the variables were flipped when it gets printed out. This is now fixed in #268 

From looking at the data files you sent, I also realized a bit about what was an issue with the fits you are making - basically, because of how you are computing the power spectra over these very wide frequency ranges, you are ending up with a very low frequency resolution (which is what the warning is about). With a low frequency resolution, it's difficult to properly fit peaks. Also, across the very wide range you were looking at, the model can't properly fit the data. If you are still looking at this data, you will need to update your spectral estimation to get higher frequency resolution, and choose narrower ranges to try and fit the model to.
